hub/api/dataset.py

```diff
@@ -128,6 +129,8 @@ def __getitem__(
             if item not in self.tensors:
                 raise TensorDoesNotExistError(item)
             else:
+                if self.index.is_trivial():
+                    return self.tensors[item]
```

Comment:
Doesn't self.tensors[item][self.index] do the same thing? Does this if condition help us in some way?

Reply:
When iterating through a tensor, we also iterate through the chunk id encoder to prefetch the chunk id and local sample index for the sub tensors, which are cached in the `Tensor._sample` attribute. If the index is trivial, then passing the base tensor instead of `Tensor[Index()]` will retain that cache.